router/module.go

```diff
@@ -278,10 +307,29 @@ func (am AppModule) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, re
 
 // OnAcknowledgementPacket implements the IBCModule interface
 func (am AppModule) OnAcknowledgementPacket(ctx sdk.Context, packet channeltypes.Packet, acknowledgement []byte, relayer sdk.AccAddress) error {
-	return am.app.OnAcknowledgementPacket(ctx, packet, acknowledgement, relayer)
+	if err := am.app.OnAcknowledgementPacket(ctx, packet, acknowledgement, relayer); err != nil {
+		return err
+	}
+
+	var ackErr channeltypes.Acknowledgement_Error
+	if err := json.Unmarshal(acknowledgement, &ackErr); err == nil && len(ackErr.Error) > 0 {
```

Comment:
Is there a reason you prefer this over:

```
	var ack channeltypes.Acknowledgement
	if err := json.Unmarshal(acknowledgement, &ack); err != nil {
		return sdkerrors.Wrapf(...)
	}

	if !ack.Success() { ... } 
```

Not a strong preference, but I like using the interface as a way of future-proofing this.

Reply:
That seems like a better way to do it, will update

Reply:
@agouin word of warning. I had to change this implementation because it was completely failing on our testnet (even if it worked fine in tests and e2e tests). Might be better to stick with what you have or at least double check this. I'll try to put together a test on top of ibc-go to figure out what was happening.